pkg/local_object_storage/metabase/exists.go

```diff
@@ -143,18 +143,18 @@ func objectStatus(tx *bbolt.Tx, metaCursor *bbolt.Cursor, addr oid.Address, curr
 
 // inGraveyard is an easier to use version of inGraveyardWithKey for cases
 // where a single address needs to be checked.
-func inGraveyard(tx *bbolt.Tx, metaCursor *bbolt.Cursor, addr oid.Address) uint8 {
+func inGraveyard(tx *bbolt.Tx, currEpoch uint64, metaCursor *bbolt.Cursor, addr oid.Address) uint8 {
 	var (
 		addrKey              = addressKey(addr, make([]byte, addressKeySize))
 		garbageContainersBkt = tx.Bucket(garbageContainersBucketName)
 		garbageObjectsBkt    = tx.Bucket(garbageObjectsBucketName)
 		graveyardBkt         = tx.Bucket(graveyardBucketName)
 	)
-	return inGraveyardWithKey(metaCursor, addrKey, graveyardBkt, garbageObjectsBkt, garbageContainersBkt)
+	return inGraveyardWithKey(metaCursor, currEpoch, addrKey, graveyardBkt, garbageObjectsBkt, garbageContainersBkt)
 }
 
-func inGraveyardWithKey(metaCursor *bbolt.Cursor, addrKey []byte, graveyard, garbageObjectsBCK, garbageContainersBCK *bbolt.Bucket) uint8 {
-	if associatedWithTypedObject(metaCursor, oid.ID(addrKey[cid.Size:]), objectSDK.TypeTombstone) {
+func inGraveyardWithKey(metaCursor *bbolt.Cursor, currEpoch uint64, addrKey []byte, graveyard, garbageObjectsBCK, garbageContainersBCK *bbolt.Bucket) uint8 {
+	if associatedWithTypedObject(currEpoch, metaCursor, oid.ID(addrKey[cid.Size:]), objectSDK.TypeTombstone) {
```

Comment:
But tombstones are set forever. Even if they expire, removed object is removed object. I don't think we need it for tombstones.